eth/filters/api.go

```diff
@@ -435,7 +435,10 @@ func (api *FilterAPI) GetFilterChanges(id rpc.ID) (interface{}, error) {
 		if !f.deadline.Stop() {
 			// timer expired but filter is not yet removed in timeout loop
 			// receive timer value and reset timer
-			<-f.deadline.C
+			select {
+			case <-f.deadline.C:
+			default:
+			}
```

Comment:
this could indeed deadlock in case there is wrong programming somewhere else. It might be more interesting to check other listeners of `f.deadline` and make sure they stop listening before the call `f.deadline.Stop()`? Same for the `pingTimer` changed in rpc/weboscket.go 🤔